go-controller/pkg/metrics/master.go

```diff
@@ -477,99 +507,239 @@ const (
 	logicalSwitchPort
 	// port binding seen in OVN-Kubernetes control plane southbound database libovsdb cache
 	portBinding
-	// port binding with updated chassis seen in OVN-Kubernetes  control plane southbound database libovsdb cache
+	// port binding with updated chassis seen in OVN-Kubernetes control plane southbound database libovsdb cache
 	portBindingChassis
-	portBindingTable = "Port_Binding"
-
-	queueCheckPeriod           = time.Millisecond * 500
-	addPortBinding   operation = iota
+	// queue operations
+	addPortBinding operation = iota
 	updatePortBinding
 	addPod
 	cleanPod
 	addLogicalSwitchPort
+	addNBGlobal
+	updateNBGlobal
+	// node limit to prevent SB DB pressure with large node counts
+	nodeLimit        = 100
+	queueCheckPeriod = time.Millisecond * 50
+	// prevent OOM by limiting queue size
+	queueLimit = 10000
+	// period to check cluster size and therefore, if we should resize ticker duration to determine availability
+	nodeCheckPeriod = 1 * time.Hour
+	delayPerNode    = 1 * time.Second
+	stopMsg         = "Stopping config duration monitor due to node count greater than %d " +
+		"to protect southbound database from overload"
+	portBindingTable = "Port_Binding"
+	nbGlobalTable    = "NB_Global"
+	// kind names
+	PodKindName     = "pod"
+	ServiceKindName = "service"
 )
 
 type record struct {
 	timestamp time.Time
 	timestampType
 }
 
-// item is the data structure for ControlPlaneRecorders queue
+type podEvent struct {
+	timestamp time.Time
+	old       model.Model
+	new       model.Model
+	uid       kapimtypes.UID
+}
+
+type configEvent struct {
+	kindStartTimestamp int
+	txStartTimestamp   int
+	kind               string
+	cfg                int
+}
+
 type item struct {
-	op     operation
-	t      time.Time
-	old    model.Model
-	new    model.Model
-	podUID kapimtypes.UID
+	op operation
+	p  *podEvent
+	c  *configEvent
 }
 
 type ControlPlaneRecorder struct {
+	// pod monitor
```

Comment:
what do you think about splitting ControlPlaneRecorder into something like PodRecorder and ConfigDurationRecorder?
and you can probably use a simple channel instead of a workqueue, since this recorder doesn't use all the extra features of a workqueue

Reply:
Totally - will do.